compiler/src/iree/compiler/Codegen/SPIRV/KernelConfig.cpp

```diff
@@ -887,9 +889,42 @@ LogicalResult setCooperativeMatrixConfig(
   tileSizes.push_back(reductionTileSizes);
   tileSizes.push_back(vectorSizes);
 
+  // Check if the C matrix will be promoted for computing shared memory usage.
+  bool promoteC = false;
+  SmallVector<Operation *> computeOps;
+  if (!failed(getComputeOps(op->getParentOfType<func::FuncOp>(), computeOps))) {
+    unsigned count = 0;
+    for (Operation *computeOp : computeOps) {
+      if (!isa<linalg::FillOp>(computeOp)) ++count;
```

Comment:
We should check whether the matmul op is having its reasult in workgroup memory. You can follow similar logic here: https://github.com/iree-org/iree/blob/main/compiler/src/iree/compiler/Codegen/SPIRV/SPIRVTileAndPromote.cpp#L319

Reply:
Since this check happens before bufferization I don't see as clean of a way to detect the result going to workgroup memory as there. I made it check whether it's stored directly to the result and assumed it would be promotion if not.

Reply:
You are right actually. Forgot about that before. Yeah this is the limitation of the existing way pipelines are done.. I'm touching promoting C in https://github.com/iree-org/iree/pull/11415. Hopefully with the changes there later we'll have a more clear estimation. So fine to have this for now!